iOS/Collection View Management /CollectionViewManager.swift

```diff
@@ -5,7 +5,9 @@ import Combine
 
 protocol ManagerSectionType: Hashable, RawRepresentable where Self.RawValue == Int {}
 
-class CollectionViewManager<SectionType: ManagerSectionType>: NSObject, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
+@MainActor
```

Comment:
Ensuring calls to collection view manager are always on the main thread.

Reply:
We actually dont need to do this. it will be more performant if we dont require it. Especially for the prefetch calls. I believe all calls to update the snapshot are upgraded to the main thread.